statsmodels/base/model.py

```diff
@@ -145,13 +146,24 @@ def from_formula(cls, formula, data, subset=None, *args, **kwargs):
         else:
             eval_env += 1  # we're going down the stack again
         missing = kwargs.get('missing', 'drop')
-        if missing == 'none':  # with patys it's drop or raise. let's raise.
+        if missing == 'none':  # with patsy it's drop or raise. let's raise.
             missing = 'raise'
 
         tmp = handle_formula_data(data, None, formula, depth=eval_env,
                                   missing=missing)
         ((endog, exog), missing_idx, design_info) = tmp
 
+        if drop_cols is not None:
+            cols = [x for x in exog.columns if x not in drop_cols]
+            exog = exog[cols]
```

Comment:
some efficiency gain, I'm not sure what makes a copy here:
make this conditional on having nothing to drop, i.e. len(cols) == len(exog.columns)  
(drop_cols = [] but there is no empty list constructed here)